### DISCUSSION TITLE
Error running the application: ModuleNotFoundError: No module named 'jwt'

### DISCUSSION THREAD

**Discussion by TalentumSebasG** [flet-dev/flet#3893] [Q&A]
### Question

![image](https://github.com/user-attachments/assets/0f1494b1-0770-4168-993b-0a6b0ecd3698)

-----------------------------------------------------------------------------------------------------------------------------------------

Traceback (most recent call last):
  File "<string>", line 43, in <module>
  File "<frozen runpy>", line 229, in run_module
  File "<frozen runpy>", line 88, in _run_code
  File "C:\Users\power\AppData\Local\Temp\serious_python_temp31193778\main.py", line 4, in <module>
  File "C:\Users\power\AppData\Local\Temp\serious_python_temp31193778\db\token.py", line 1, in <module>
ModuleNotFoundError: No module named 'jwt'

-----------------------------------------------------------------------------------------------------------------------------------------

It seems that the error is due to the jwt module not being found.

In my development environment, the application works perfectly, and I have the jwt package installed. However, when I generate the build for Windows using flutter build windows, it appears that this package is not included in the resulting environment.

Could anyone help me understand why the jwt module is not found in the Windows build and how I can resolve this?

-----------------------------------------------------------------------------------------------------------------------------------------

Parece que el error se debe a que no se puede encontrar el módulo jwt. 

En mi ambiente de desarrollo, la aplicación funciona perfectamente y tengo instalado el paquete jwt. Sin embargo, cuando genero el build para Windows utilizando flutter build windows, parece que no incluye ese paquete en el entorno resultante.

¿Alguien podría ayudarme a entender por qué no se encuentra el módulo jwt en el build para Windows y cómo puedo solucionarlo?

### Error message

```bash
Traceback (most recent call last):
  File "<string>", line 43, in <module>
  File "<frozen runpy>", line 229, in run_module
  File "<frozen runpy>", line 88, in _run_code
  File "C:\Users\power\AppData\Local\Temp\serious_python_temp31193778\main.py", line 4, in <module>
  File "C:\Users\power\AppData\Local\Temp\serious_python_temp31193778\db\token.py", line 1, in <module>
ModuleNotFoundError: No module named 'jwt'
```

### ------------------------------------------------------

- [x] I have searched for answers to my question both in the issues and in previous discussions.

**Comment by ndonkoHenri** [comment]
Did you add it to the requirements file?

**Reply by TalentumSebasG** [reply]
I initially did not have a requirements.txt file in my project, so I created one using the command pip freeze > requirements.txt to capture all the installed dependencies. However, when I tried to run flutter build windows, I encountered an error message stating that the file was not created correctly or was not found by the system.

![image](https://github.com/user-attachments/assets/6bb9307f-1eb1-4e17-8492-4937280ac720)

**Reply by TalentumSebasG** [reply]
i dont have issues with flutter

![image](https://github.com/user-attachments/assets/024822ea-8af5-444d-ad7a-e647e1a4fc6c)

**Reply by ndonkoHenri** [reply]
It is recommended not to use pip freeze, but to hand-pick the libs you need. So if you use pillow, add just pillow to the requirements file.
Also, upgrade to Flet v0.23.2, which is the latest version atm, and try rebuilding your app this time in [verbose](http://localhost:3000/docs/publish/#verbose-logging) mode, then share the logs here.

**Reply by TalentumSebasG** [reply]
This is the verbose. 

`PS E:\Proyectos\python escritorio\flet\flet_pos> flet build windows --verbose
[17:02:18] Flutter bootstrap directory: C:\Users\power\AppData\Local\Temp\flet_flutter_build_UbPcxzt5qh
           Additional Flutter dependencies:
           {}
[17:02:21] Created Flutter bootstrap project from gh:flet-dev/flet-build-template with ref 0.23.2 ✅
           Customized app icons and splash images ✅
           Run subprocess: ['C:\\dev\\flutter\\bin\\dart.BAT', 'run', 'flutter_launcher_icons']
(   ●  ) Generating app icons ⏳... The Dart CLI developer tool uses Google Analytics to report usage and diagnostic
data along with package dependencies, and crash reporting to send basic crash
reports. This data is used to help improve the Dart platform, Flutter framework,
and related tools.

Telemetry is not sent on the very first run. To disable reporting of telemetry,
run this terminal command:

    dart --disable-analytics

If you opt out of telemetry, an opt-out event will be sent, and then no further
information will be sent. This data is collected in accordance with the Google
Privacy Policy (https://policies.google.com/privacy).

(     ●) Generating app icons (2.0s)
(    ● ) Generating app icons ⏳... 
Got dependencies in `C:\Users\power\AppData\Local\Temp\flet_flutter_build_UbPcxzt5qh`.
( ●    ) Generating app icons ⏳... 
Built flutter_launcher_icons:flutter_launcher_icons.
( ●    ) Generating app icons ⏳...   ════════════════════════════════════════════
     FLUTTER LAUNCHER ICONS (v0.13.1)
  ════════════════════════════════════════════
  
(  ●   ) Generating app icons ⏳... • Creating default icons Android
(   ●  ) Generating app icons ⏳... • Overwriting the default Android launcher icon with a new icon
(   ●  ) Generating app icons ⏳... • Overwriting the default Android launcher icon with a new icon
(   ●  ) Generating app icons ⏳... • Overwriting default iOS launcher icon with new icon
(   ●  ) Generating app icons ⏳... Creating Icons for Web...
(  ●   ) Generating app icons ⏳... Creating Icons for Windows...
(   ●  ) Generating app icons ⏳... Creating Icons for Web...
(  ●   ) Generating app icons ⏳... Creating Icons for Windows...
(   ●  ) Generating app icons ⏳... Creating Icons for MacOS...
(  ●   ) Generating app icons ⏳... 
(  ●   ) Generating app icons ⏳... 
✓ Successfully generated launcher icons
[17:02:28] Generated app icons ✅
[17:02:28] Generated app icons ✅
           Run subprocess: ['C:\\dev\\flutter\\bin\\dart.BAT', 'run', 'serious_python:main', 'package', 'E:\\Proyectos\\python escritorio\\flet\\flet_pos',      
           '--dep-mappings', 'flet=flet-embed', '--req-deps', 'flet-embed', '--exclude', 'build']
(  ●   ) Packaging Python app ⏳... 
Built serious_python:main.
(  ●   ) Packaging Python app ⏳... Running package command
(   ●  ) Packaging Python app ⏳... Creating asset directory: C:\Users\power\AppData\Local\Temp\flet_flutter_build_UbPcxzt5qh\app
Copying Python app from E:\Proyectos\python escritorio\flet\flet_pos to C:\Users\power\AppData\Local\Temp\serious_python_temp4dcef9d0
(    ● ) Packaging Python app ⏳... Error: FileSystemException: Failed to decode data using encoding 'utf-8', path = 'C:\Users\power\AppData\Local\Temp\serious_python_temp4dcef9d0\requirements.txt'
Deleting temp directory C:\Users\power\AppData\Local\Temp\serious_python_temp4dcef9d0
[17:02:30] Deleting Flutter bootstrap directory C:\Users\power\AppData\Local\Temp\flet_flutter_build_UbPcxzt5qh
           Flet app package app/app.zip was not created.`

**Reply by ndonkoHenri** [reply]
> Error: FileSystemException: Failed to decode data using encoding 'utf-8', path = 'C:\Users\power\AppData\Local\Temp\serious_python_temp4dcef9d0\requirements.txt'

Your requirements file must be utf-8 encoded. Either delete and recreate with the proper encoding or search the internet on how to modify it's encoding.

**Reply by TalentumSebasG** [reply]
To resolve the issue with the requirements.txt file, I first saved the file using the Windows 11 text editor. Initially, the file might have been saved in an encoding that is not compatible with the packaging process. To ensure the file was saved correctly, I changed its encoding to UTF-8, which is the recommended standard for text files. This action should ensure that the file can be read correctly during the packaging process and avoid any encoding-related errors.

![image](https://github.com/user-attachments/assets/01abaa20-5d84-44bf-990b-af7f23b91409)

PS E:\Proyectos\python_desktop\flet\flet_pos> flet build windows --verbose
[18:32:06] Flutter bootstrap directory: C:\Users\power\AppData\Local\Temp\flet_flutter_build_MDxPliM5Rq
           Additional Flutter dependencies:
           {}
[18:32:10] Created Flutter bootstrap project from gh:flet-dev/flet-build-template with ref 0.23.2 ✅
           Customized app icons and splash images ✅
           Run subprocess: ['C:\\dev\\flutter\\bin\\dart.BAT', 'run', 'flutter_launcher_icons']
(    ● ) Generating app icons (1.7s)
Downloading packages... 
Got dependencies in `C:\Users\power\AppData\Local\Temp\flet_flutter_build_MDxPliM5Rq`.
(     ●) Generating app icons ⏳... 
Built flutter_launcher_icons:flutter_launcher_icons.
( ●    ) Generating app icons ⏳...   ════════════════════════════════════════════
     FLUTTER LAUNCHER ICONS (v0.13.1)
  ════════════════════════════════════════════
  
(●     ) Generating app icons ⏳... • Creating default icons Android
( ●    ) Generating app icons ⏳... • Overwriting the default Android launcher icon with a new icon
(    ● ) Generating app icons ⏳... • Overwriting default iOS launcher icon with new icon
(    ● ) Generating app icons ⏳... Creating Icons for Web...
( ●    ) Generating app icons ⏳... Creating Icons for Windows...
( ●    ) Generating app icons ⏳... Creating Icons for MacOS...
( ●    ) Generating app icons ⏳... 
✓ Successfully generated launcher icons
[18:32:16] Generated app icons ✅
           Run subprocess: ['C:\\dev\\flutter\\bin\\dart.BAT', 'run', 'serious_python:main', 'package', 'E:\\Proyectos\\python_desktop\\flet\\flet_pos',    
           '--dep-mappings', 'flet=flet-embed', '--req-deps', 'flet-embed', '--exclude', 'build']
(  ●   ) Packaging Python app ⏳... 
Built serious_python:main.
(   ●  ) Packaging Python app ⏳... Running package command
Creating asset directory: C:\Users\power\AppData\Local\Temp\flet_flutter_build_MDxPliM5Rq\app
Copying Python app from E:\Proyectos\python_desktop\flet\flet_pos to C:\Users\power\AppData\Local\Temp\serious_python_tempe4595b5d
(     ●) Packaging Python app ⏳... Installing dependencies [flet-embed==0.23.2, fpdf==1.7.2, PyJWT==2.8.0] with pip command to C:\Users\power\AppData\Local\Temp\serious_python_tempe4595b5d\__pypackages__
(    ● ) Packaging Python app ⏳... Extracting Python distributive from C:\Users\power\AppData\Local\Temp\cpython-3.11.6+20231002-x86_64-pc-windows-msvc-shared-install_only.tar.gz to C:\Users\power\AppData\Local\Temp\hostpython3.11_c90dad88
(●     ) Packaging Python app ⏳... Compiling Python sources at C:\Users\power\AppData\Local\Temp\serious_python_tempe4595b5d
(     ●) Packaging Python app ⏳... Delete unnecessary files with extensions: [.py, .c, .h, .typed, .exe]
Delete unnecessary files and directories: [__pycache__, bin]
(  ●   ) Packaging Python app ⏳... Creating app archive at C:\Users\power\AppData\Local\Temp\flet_flutter_build_MDxPliM5Rq\app/app.zip from C:\Users\power\(   ●  ) Packaging Python app ⏳... 
(   ●  ) Packaging Python app ⏳... Deleting temp directory C:\Users\power\AppData\Local\Temp\serious_python_tempe4595b5d
(     ●) Packaging Python app ⏳... Deleting Python directory C:\Users\power\AppData\Local\Temp\hostpython3.11_c90dad88
[18:32:47] Packaged Python app ✅
           Run subprocess: ['C:\\dev\\flutter\\bin\\flutter.BAT', 'build', 'windows']
( ●    ) Building Windows app ⏳... 
(  ●   ) Building Windows app ⏳... 
(    ● ) Building Windows app ⏳...   collection 1.18.0 (1.19.0 available)
  fl_chart 0.65.0 (0.69.0 available)
  flet 0.23.2 (0.24.0 available)
  flutter_lints 2.0.3 (4.0.0 available)
  flutter_markdown 0.6.23 (0.7.3+1 available)
  http_parser 4.0.2 (4.1.0 available)
  js 0.6.7 (0.7.1 available)
  lints 2.1.1 (4.0.0 available)
  material_color_utilities 0.8.0 (0.12.0 available)
  meta 1.12.0 (1.15.0 available)
! package_info_plus 8.0.2 (overridden)
  screen_retriever 0.1.9 (0.2.0 available)
  sensors_plus 4.0.2 (6.0.1 available)
  sensors_plus_platform_interface 1.2.0 (2.0.0 available)
  toml 0.15.0 (0.16.0 available)
  url_launcher_android 6.3.9 (6.3.10 available)
  url_strategy 0.2.0 (0.3.0 available)
! wakelock_plus 1.2.8 (overridden)
  web_socket_channel 2.4.0 (3.0.1 available)
  window_manager 0.3.9 (0.4.2 available)
Got dependencies!
18 packages have newer versions incompatible with dependency constraints.
Try `flutter pub outdated` for more information.
( ●    ) Building Windows app ⏳... 
windows/flutter/CMakeLists.txt does not use FLUTTER_TARGET_PLATFORM, updating.
C:\Program Files\Microsoft Visual Studio\2022\Community\MSBuild\Microsoft\VC\v170\Microsoft.CppBuild.targets(541,5): warning MSB8029: El directorio intermedio o el directorio de salida no pueden estar en el directorio temporal, ya que podría dar lugar a problemas con la compilación incremental. [C:\Users\power\AppData\Local\Temp\flet_flutter_build_MDxPliM5Rq\build\windows\x64\ZERO_CHECK.vcxproj]
C:\Program Files\Microsoft Visual Studio\2022\Community\MSBuild\Microsoft\VC\v170\Microsoft.CppBuild.targets(541,5): warning MSB8029: El directorio intermedio o el directorio de salida no pueden estar en el directorio temporal, ya que podría dar lugar a problemas con la compilación incremental. [C:\Users\power\AppData\Local\Temp\flet_flutter_build_MDxPliM5Rq\build\windows\x64\plugins\serious_python_windows\CopyPythonDLLs.vcxproj]
C:\Program Files\Microsoft Visual Studio\2022\Community\MSBuild\Microsoft\VC\v170\Microsoft.CppBuild.targets(541,5): warning MSB8029: El directorio intermedio o el directorio de salida no pueden estar en el directorio temporal, ya que podría dar lugar a problemas con la compilación incremental. [C:\Users\power\AppData\Local\Temp\flet_flutter_build_MDxPliM5Rq\build\windows\x64\flutter\flutter_assemble.vcxproj]
C:\Program Files\Microsoft Visual Studio\2022\Community\MSBuild\Microsoft\VC\v170\Microsoft.CppBuild.targets(541,5): warning MSB8029: El directorio intermedio o el directorio de salida no pueden estar en el directorio temporal, ya que podría dar lugar a problemas con la compilación incremental. [C:\Users\power\AppData\Local\Temp\flet_flutter_build_MDxPliM5Rq\build\windows\x64\flutter\flutter_wrapper_app.vcxproj]
C:\Program Files\Microsoft Visual Studio\2022\Community\MSBuild\Microsoft\VC\v170\Microsoft.CppBuild.targets(541,5): warning MSB8029: El directorio intermedio o el directorio de salida no pueden estar en el directorio temporal, ya que podría dar lugar a problemas con la compilación incremental. [C:\Users\power\AppData\Local\Temp\flet_flutter_build_MDxPliM5Rq\build\windows\x64\flutter\flutter_wrapper_plugin.vcxproj]
C:\Program Files\Microsoft Visual Studio\2022\Community\MSBuild\Microsoft\VC\v170\Microsoft.CppBuild.targets(541,5): warning MSB8029: El directorio intermedio o el directorio de salida no pueden estar en el directorio temporal, ya que podría dar lugar a problemas con la compilación incremental. [C:\Users\power\AppData\Local\Temp\flet_flutter_build_MDxPliM5Rq\build\windows\x64\plugins\screen_retriever\screen_retriever_plugin.vcxproj]
C:\Program Files\Microsoft Visual Studio\2022\Community\MSBuild\Microsoft\VC\v170\Microsoft.CppBuild.targets(541,5): warning MSB8029: El directorio intermedio o el directorio de salida no pueden estar en el directorio temporal, ya que podría dar lugar a problemas con la compilación incremental. [C:\Users\power\AppData\Local\Temp\flet_flutter_build_MDxPliM5Rq\build\windows\x64\plugins\serious_python_windows\serious_python_windows_plugin.vcxproj]
C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.41.34120\include\format(42): warning STL4038: The contents of <format> are available only with C++20 or later. [C:\Users\power\AppData\Local\Temp\flet_flutter_build_MDxPliM5Rq\build\windows\x64\plugins\serious_python_windows\serious_python_windows_plugin.vcxproj]
C:\Program Files\Microsoft Visual Studio\2022\Community\MSBuild\Microsoft\VC\v170\Microsoft.CppBuild.targets(541,5): warning MSB8029: El directorio intermedio o el directorio de salida no pueden estar en el directorio temporal, ya que podría dar lugar a problemas con la compilación incremental. [C:\Users\power\AppData\Local\Temp\flet_flutter_build_MDxPliM5Rq\build\windows\x64\plugins\url_launcher_windows\url_launcher_windows_plugin.vcxproj]
C:\Program Files\Microsoft Visual Studio\2022\Community\MSBuild\Microsoft\VC\v170\Microsoft.CppBuild.targets(541,5): warning MSB8029: El directorio intermedio o el directorio de salida no pueden estar en el directorio temporal, ya que podría dar lugar a problemas con la compilación incremental. [C:\Users\power\AppData\Local\Temp\flet_flutter_build_MDxPliM5Rq\build\windows\x64\plugins\window_manager\window_manager_plugin.vcxproj]
C:\Program Files\Microsoft Visual Studio\2022\Community\MSBuild\Microsoft\VC\v170\Microsoft.CppBuild.targets(541,5): warning MSB8029: El directorio intermedio o el directorio de salida no pueden estar en el directorio temporal, ya que podría dar lugar a problemas con la compilación incremental. [C:\Users\power\AppData\Local\Temp\flet_flutter_build_MDxPliM5Rq\build\windows\x64\plugins\window_to_front\window_to_front_plugin.vcxproj]
C:\Program Files\Microsoft Visual Studio\2022\Community\MSBuild\Microsoft\VC\v170\Microsoft.CppBuild.targets(541,5): warning MSB8029: El directorio intermedio o el directorio de salida no pueden estar en el directorio temporal, ya que podría dar lugar a problemas con la compilación incremental. [C:\Users\power\AppData\Local\Temp\flet_flutter_build_MDxPliM5Rq\build\windows\x64\runner\flet_pos.vcxproj]
C:\Program Files\Microsoft Visual Studio\2022\Community\MSBuild\Microsoft\VC\v170\Microsoft.CppBuild.targets(541,5): warning MSB8029: El directorio intermedio o el directorio de salida no pueden estar en el directorio temporal, ya que podría dar lugar a problemas con la compilación incremental. [C:\Users\power\AppData\Local\Temp\flet_flutter_build_MDxPliM5Rq\build\windows\x64\ALL_BUILD.vcxproj]
C:\Program Files\Microsoft Visual Studio\2022\Community\MSBuild\Microsoft\VC\v170\Microsoft.CppBuild.targets(541,5): warning MSB8029: El directorio intermedio o el directorio de salida no pueden estar en el directorio temporal, ya que podría dar lugar a problemas con la compilación incremental. [C:\Users\power\AppData\Local\Temp\flet_flutter_build_MDxPliM5Rq\build\windows\x64\INSTALL.vcxproj]
(    ● ) Building Windows ap  160,3s
√ Built build\windows\x64\runner\Release\flet_pos.exe
[18:35:30] Built Windows app ✅
           Copying build output from: C:\Users\power\AppData\Local\Temp\flet_flutter_build_MDxPliM5Rq\build\windows\x64\runner\Release\*
[18:35:34] Copied build to build\windows directory ✅
           Deleting Flutter bootstrap directory C:\Users\power\AppData\Local\Temp\flet_flutter_build_MDxPliM5Rq
[18:35:37] Successfully built your Windows app! 🥳 Find it in build\windows directory. 📂
